### ISSUE TITLE
Error with reflect-metadata and zone when upgrading to beta.0

### ISSUE THREAD

**Issue by hitmanmcc** [angular/angular#5958]
Hi all!

I'm trying to upgrade to the newest angular2 version (beta.0). Initially I had problems with rxjs and zonejs, eventually I sorted those but now I'm stuck with an error regarding reflect-metadata shim. The error reads as follows:
```js
reflect-metadata shim is required when using class decorators
```
At this stage my code is already transpiled so I don't really understand why I'm getting this error.
To get around this I tried to import zone and reflect-metadata in my bootstrap.js file and, add the following map to the System.config file, which reads:
```js
System.config({
      map: {
         'zone.js': '../node_modules/zone.js/dist/zone.min.js',
         'reflect-metadata': '../node_modules/reflect-metadata/Reflect.js'
      }
    });
```

This results in another error:
```js
Error: XHR error (404 Not Found) loading http://localhost:12345/build/crypto(…)
```
How do I solve this?

P.S. - Congrats on the beta.

**Comment by pkozlowski-opensource** [comment]
@hitmanmcc please go over breaking changes in https://github.com/angular/angular/blob/master/CHANGELOG.md

In short: don't load zones nor reflect-metadata through system loader. Include `angular2-polyfills.js` using a script tag instead. More info in https://github.com/angular/angular/blob/master/modules/angular2/docs/bundles/overview.md 

You might check the working beta.0 setup with SystemJS here:
https://github.com/pkozlowski-opensource/ng2-play/blob/master/index.html

**Comment by hitmanmcc** [comment]
@pkozlowski-opensource Thanks for the quick reply, I did indeed read the breaking changes, I always read the whole changelog. We've been working with angular 2 since the very first alphas all the way up until now. :)
Since this time we didn't immediately update to the following versions, I guess I must have missed that. We stopped at alpha 50 and now we were trying to upgrade to beta.0.
Thank you very much for your quick reply and for the additional links and information. A working setup was also something I was trying to find before having posted this.
